### PR TITLE
iwd: Simpler handling of wifi secrets

### PR DESCRIPTION
**Pull request by balrog-kun**
I'll appreciate a thorough look at the changes as I'm not 100% sure that for example changing the NMDevice state to NEED_AUTH and back to CONFIG from the subclass won't break anything.  I'm also not sure whether nm_connection_clear_secrets() actually does anything without some other call to make the change be saved to file (doesn't really matter as I'm setting the NM_SECRET_AGENT_GET_SECRETS_FLAG_REQUEST_NEW flag anyway, but this could be avoided)

### CODE REVIEW

**Comment by vhumpa** [comment]
Can one of the admins verify this patch?

File: src/devices/wifi/nm-device-iwd.c [hunk 1]
```diff
@@ -1881,31 +1753,44 @@ nm_device_iwd_set_dbus_object (NMDeviceIwd *self, GDBusObject *object)
...
 
-	return nm_setting_wireless_security_get_psk (s_wireless_sec);
+	nm_device_state_changed (NM_DEVICE (self), NM_DEVICE_STATE_NEED_AUTH,
+	                         NM_DEVICE_STATE_REASON_NO_SECRETS);
+
+	if (!wifi_secrets_get_secrets (self,
+	                               NM_SETTING_WIRELESS_SECURITY_SETTING_NAME,
+	                               NM_SECRET_AGENT_GET_SECRETS_FLAG_ALLOW_INTERACTION
+		                         | NM_SECRET_AGENT_GET_SECRETS_FLAG_REQUEST_NEW))
+		return FALSE;
```
Comment (thom311) [inline src/devices/wifi/nm-device-iwd.c]: as said above, `wifi_secrets_get_secrets` cannot fail.

File: src/devices/wifi/nm-device-iwd.c [hunk 2]
```diff
@@ -1881,31 +1753,44 @@ nm_device_iwd_set_dbus_object (NMDeviceIwd *self, GDBusObject *object)
...
+	psk = nm_setting_wireless_security_get_psk (s_wireless_sec);
+	if (psk) {
+		_LOGD (LOGD_DEVICE | LOGD_WIFI,
+		       "Returning the PSK to the IWD Agent");
 
-	s_wireless_sec = nm_connection_get_setting_wireless_security (connection);
-	if (!s_wireless_sec)
-		return NULL;
+		g_dbus_method_invocation_return_value (invocation,
+	                                               g_variant_new ("(s)", psk));
```
Comment (thom311) [inline src/devices/wifi/nm-device-iwd.c]: indentation (tab vs. space)

**Comment by balrog-kun** [comment]
Thanks for the check, updated the patch.

**Comment by balrog-kun** [comment]
Updated once again with a minor change to account for the new NEED_AUTH state meaning.

**Comment by thom311** [comment]
Obsoleted by #73

**Pull request closed without merging**